sdk/python/express_relay/svm/limo_client.py

```diff
@@ -111,82 +107,72 @@ def create_associated_token_account_idempotent(
             data=bytes([1]),  # idempotent version of the instruction
         )
 
-    async def get_ata_and_create_ixn_if_required(
+    def get_ata_and_create_ixn_if_required(
         self,
         owner: Pubkey,
         token_mint_address: Pubkey,
         token_program_id: Pubkey,
         payer: Pubkey,
     ) -> Tuple[Pubkey, Sequence[Instruction]]:
         ata = self.get_ata(owner, token_mint_address, token_program_id)
-        if not await self.account_exists(ata):
-            ix = self.create_associated_token_account_idempotent(
-                payer, owner, token_mint_address, token_program_id
-            )
-            return ata, [ix]
-        return ata, []
+        ix = self.create_associated_token_account_idempotent(
+            payer, owner, token_mint_address, token_program_id
+        )
+        return ata, [ix]
 
-    async def get_init_if_needed_wsol_create_and_close_ixs(
+    def get_init_if_needed_wsol_create_and_close_ixs(
         self, owner: Pubkey, payer: Pubkey, amount_to_deposit_lamports: int
     ) -> WSOLInstructions:
         """
         Returns necessary instructions to create, fill and close a wrapped SOL account.
-        If the account already exists:
-            it makes sure the balance is at least `amount_to_deposit_lamports` with no create or close instructions.
-        If the account does not exist:
-            it creates the account, fills it with `amount_to_deposit_lamports` lamports and close it in the end.
+        Creation instruction is idempotent.
+        Filling instruction doesn't take into account the current WSOL balance.
+        Closing instruction unwraps all the WSOL back to the owner, even if it was deposited by another transaction.
```

Comment:
nit: "even if it was deposited by another transaction" is a bit confusing, i think you can just say the WSOL token account is always closed and the SOL is unwrapped back to the owner